autox/agent/ddpg/ddpg.py

```diff
@@ -0,0 +1,263 @@
+import os
+
+import numpy as np
+
+import torch
+import torch.nn as nn
+from torch.optim import Adam
+
+from autox.agent.ddpg.memory import SequentialMemory
+from autox.utils.utils import to_numpy, to_tensor, sample_from_truncated_normal_distribution
+
+from types import SimpleNamespace
+
+criterion = nn.MSELoss()
+USE_CUDA = torch.cuda.is_available()
+
+class Actor(nn.Module):
+    def __init__(self, nb_states, nb_actions, hidden1=400, hidden2=300, init_w=3e-3):
+        super(Actor, self).__init__()
+        self.fc1 = nn.Linear(nb_states, hidden1)
+        self.fc2 = nn.Linear(hidden1, hidden2)
+        self.fc3 = nn.Linear(hidden2, nb_actions)
+        self.relu = nn.ReLU()
+        self.sigmoid = nn.Sigmoid()
+
+    def forward(self, x):
+        out = self.fc1(x)
+        out = self.relu(out)
+        out = self.fc2(out)
+        out = self.relu(out)
+        out = self.fc3(out)
+        out = self.sigmoid(out)
+        return out
+
+
+class Critic(nn.Module):
+    def __init__(self, nb_states, nb_actions, hidden1=400, hidden2=300, init_w=3e-3):
+        super(Critic, self).__init__()
+        self.fc11 = nn.Linear(nb_states, hidden1)
+        self.fc12 = nn.Linear(nb_actions, hidden1)
+        self.fc2 = nn.Linear(hidden1, hidden2)
+        self.fc3 = nn.Linear(hidden2, 1)
+        self.relu = nn.ReLU()
+
+    def forward(self, xs):
+        x, a = xs
+        out = self.fc11(x) + self.fc12(a)
+        out = self.relu(out)
+        out = self.fc2(out)
+        out = self.relu(out)
+        out = self.fc3(out)
+        return out
+
+
+class DDPG(object):
```

Comment:
We have to be sure, that this implementation of `DDPG`, `EpisodeParameterMemory` and `SequentialMemory` are functionally correct.

Please add unit tests for the key methods, like  `select_action` and `update_policy` for DDPG, `sample` and `append` for Memory-like classes.

Test scenario might be the following: 
- initialize a tested class by something simple, for which it is easy to calculate reference manually or using simple reference code.
- run a tested method for the class instance, get the actual output
- compare the actual output with predefined reference output

Existing tests as an example: [1](https://github.com/openvinotoolkit/nncf/blob/develop/tests/sparsity/rb/test_weights.py#L58), [2](https://github.com/openvinotoolkit/nncf/blob/develop/tests/quantization/test_range_init.py#L691), [3](https://github.com/openvinotoolkit/nncf/blob/develop/tests/quantization/test_quantizer_propagation_solver.py#L1107)

Reply:
Low priority. Please consider adding tests for interface methods.